compiler/src/dotty/tools/dotc/core/TypeErrors.scala

```diff
@@ -40,16 +40,22 @@ class MissingType(pre: Type, name: Name) extends TypeError {
   }
 }
 
-class RecursionOverflow(val op: String, details: => String, previous: Throwable, val weight: Int) extends TypeError {
+class RecursionOverflow(val op: String, details: => String, val previous: Throwable, val weight: Int) extends TypeError {
 
   def explanation: String = s"$op $details"
 
   private def recursions: List[RecursionOverflow] = {
-    val nested = previous match {
-      case previous: RecursionOverflow => previous.recursions
-      case _ => Nil
+    import scala.collection.mutable.ListBuffer
+    val result = ListBuffer.empty[RecursionOverflow]
+    @annotation.tailrec def loop(throwable: Throwable): List[RecursionOverflow] = throwable match {
+      case ro: RecursionOverflow =>
+        if (result.contains(ro)) println("caught the issue")
```

Comment:
It’d be very surprising if the `RecursionOverflow` chain was cyclic. However, the length of the chain is proportional to the stack depth, so it’s best traversed in constant stack size. To see why, you need to see how `RecursionOverflow` is created by `handleRecursive` and how that’s used in certain recursive methods in Typer (I can explain in person if desired).

Reply:
Yeah, I wasn't sure if the `RecursionOverflow` chain can actually be cyclic. If it's really proportional to stack depth, then it makes sense to construct `recursions` in a loop. How about I simply open a separate PR with that change and we merge it on its own?

Reply:
Go ahead!